Sources/MSLFoundation/BackgroundTaskManager/BackgroundTaskManager.swift

```diff
@@ -0,0 +1,146 @@
+import BackgroundTasks
+import Combine
+import Foundation
+import Logging
+import UIKit
+
+private let logger: Logger = {
+    var logger = Logger(label: "\(#file)")
+    logger.logLevel = .info
+    return logger
+}()
+
+public enum EnqueueType {
+    case replace
+    case keep
+}
+
+public final class BackgroundTaskManager {
```

Comment:
### 💡 Suggestion, _non-blocking_

Could you please throw some comments on these things if this is what we would be expected to be calling into? It seems like this is basically the entry point into all of this code? At least maybe just the `BackgroundTaskManager` and its `register` and `unregister`?